src/codeu/chat/server/View.java

```diff
@@ -78,6 +78,81 @@ public View(Model model) {
   @Override
   public Message findMessage(Uuid id) { return model.messageById().first(id); }
 
+  @Override
+  public Collection<String> userStatusUpdate(String name, Uuid owner) {
+
+    Collection<ConversationPayload> allConversations = all(model.conversationPayloadById());
+    Collection<String> contributions = new ArrayList<String>();
+
+    final User foundOwner = model.userById().first(owner);
+    final User foundUser = model.userByText().first(name);
+
+    if(foundUser != null) {
+      if(foundOwner.UserSet.contains(foundUser.id)) {
+        final Time lastUpdate = foundOwner.UserUpdateMap.get(foundUser.id);
+        for(ConversationPayload conversationPayload : allConversations) {
+          Message currentMessage = model.messageById().first(conversationPayload.firstMessage);
+          boolean foundMessage = false;
+          while(currentMessage != null && foundMessage == false) {
+            if(lastUpdate.compareTo(currentMessage.creation) < 0 && currentMessage.author.equals(foundUser.id)) {
+              ConversationHeader conversationContribution = model.conversationById().first(conversationPayload.id);
+              contributions.add(conversationContribution.title);
+              LOG.info("Conversation added to list.");
+              foundMessage = true;
+            }
+            currentMessage = model.messageById().first(currentMessage.next);
+          }
+          if(!contributions.contains(model.conversationById().first(conversationPayload.id).title)) {
```

Comment:
The second check is to see if the user is in the current user's user interest set (i.e the specified user that was named was added to the current signed in user's interests at some point with i-u-add and can be found in user.UserSet). A user can only call status updates on users that they have added to their user interest set.